packages/core/src/stylable-resolver.ts

```diff
@@ -24,8 +24,24 @@ export type JsModule = {
     default?: unknown;
     [key: string]: unknown;
 };
-export type CachedModule = StylableMeta | JsModule | null;
-export type StylableResolverCache = Map<string, StylableMeta | JsModule | null>;
+
+interface BaseCachedModule<T> {
+    resolvedPath: string;
+    error?: unknown;
+    value: T | null;
+}
+
+export interface CachedStylableMeta extends BaseCachedModule<StylableMeta> {
+    kind: 'css';
+}
+
+export interface CachedJsModule extends BaseCachedModule<JsModule> {
+    kind: 'js';
+}
+
+export type CachedModuleEntity = CachedStylableMeta | CachedJsModule;
+export type CachedModule = CachedModuleEntity['value'];
```

Comment:
Not sure about this type (`CachedModule`), it's exported from the `index` and it is created not to break anyone who uses it.
I would really want to remove it from here and from the `index`.

@barak007